docs/source/llms/prompt-engineering.rst

```diff
@@ -0,0 +1,341 @@
+Prompts can define multiple variables. For example, you can use the following prompt template
+to instruct the LLM to answer questions about the MLflow documentation:
+
+*Read the following article from the MLflow documentation that appears between triple
+backticks. Then, answer the question about the documentation that appears between triple quotes.
+Include relevant links and code examples in your answer.*
+
+*```{{article}}```*
+
+*"""*
+*{{question}}*
+*"""*
+
+Then, fill in the input variables. For example, in the MLflow documentation
+use case, the *article* input variable can be set to the contents of
+https://mlflow.org/docs/latest/tracking.html#logging-data-to-runs and the *question* input variable
+can be set to ``"How do I create a new MLflow Run using the Python API?"``.
+
+Finally, click the **Evaluate** button to see the new output. You can also try choosing a larger
+value of *temperature* to observe how the LLM's output changes.
+
+.. figure:: ../_static/images/prompt_modal_3.png
+   :scale: 35%
+   :align: center
+
+Step 7: Capture your choice of LLM, prompt template, and parameters as an MLflow Run
+------------------------------------------------------------------------------------
+Once you're satisfied with your chosen prompt template and parameters, click the **Create Run**
+button to store this information, along with your choice of LLM, as an MLflow Run. This will
+create a new Run with the prompt template, parameters, and choice of LLM stored as Run params.
+It will also automatically create an MLflow Model with this information that can be used for batch
+or real-time inference.
+
+1. To view this information, click the Run name to open the **Run** page:
+
+    .. figure:: ../_static/images/prompt_eng_run_page.png
+       :scale: 25%
+       :align: center
+
+2. You can also see the parameters and compare them with other configurations by opening the **Table**
+   view tab:
+
+    .. figure:: ../_static/images/prompt_eng_table_view.png
+       :scale: 25%
+       :align: center
+
+3. After your Run is created, MLflow will open the **Evaluation** tab where you can see your latest
+   playground input & output and try out additional inputs:
+
+    .. figure:: ../_static/images/eval_view_1.png
+       :scale: 25%
+       :align: center
+
+
+Step 8: Try new inputs
+----------------------
+To test the behavior of your chosen LLM, prompt template, and parameters on a new inputs:
+
+1. Click the *Add Row* button and fill in a value(s) your prompt template's input variable(s).
+   For example, in the MLflow documentation use case, you can try asking a question
+   unrelated to MLflow to see how the LLM responds. This is important to ensure that the application
+   is robust to irrelevant inputs.
+
+    |add_row| |add_row_modal|
+
+    .. |add_row| image:: ../_static/images/add_row.png
+       :width: 10%
+
+    .. |add_row_modal| image:: ../_static/images/add_row_modal.png
+       :width: 50%
```

Comment:
This part looks like this. Is this intended?

<img width="1640" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/c4b96427-ac51-4d16-b881-237ba851a696">

Reply:
Yes, let me know if you have suggestions for improvements